core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java

```diff
@@ -1,11 +1,120 @@
+        private static final Pattern VERSION_PATTERN = Pattern.compile(VERS_FORMAT);
+
+        private static final Version UNKNOWN_VERSION = null;
+
+        static Version parse(List<String> lines) {
+            Matcher firstMatcher = FIRST_PATTERN.matcher(lines.get(0));
+            Matcher secondMatcher = SECOND_PATTERN.matcher(lines.get(1));
+            Matcher thirdMatcher = THIRD_PATTERN.matcher(lines.get(2));
+            if (firstMatcher.find() && secondMatcher.find() && thirdMatcher.find()) {
+                String javaVersion = firstMatcher.group(VNUM_GROUP);
+                java.lang.Runtime.Version v = null;
+                try {
+                    v = java.lang.Runtime.Version.parse(javaVersion);
+                } catch (IllegalArgumentException e) {
+                    return UNKNOWN_VERSION;
+                }
+
+                String vendorVersion = secondMatcher.group(VENDOR_VERSION_GROUP);
+
+                String buildInfo = secondMatcher.group(BUILD_INFO_GROUP);
+                String graalVersion = graalVersion(buildInfo);
+                String mandrelVersion = mandrelVersion(vendorVersion);
+                Distribution dist = isMandrel(vendorVersion) ? Distribution.MANDREL : Distribution.ORACLE;
+                String versNum = (dist == Distribution.MANDREL ? mandrelVersion : graalVersion);
+                return new Version(lines.stream().collect(Collectors.joining("\n")),
+                        versNum, v.feature(), v.update(), dist);
+            } else {
+                return UNKNOWN_VERSION;
+            }
+        }
+
+        private static boolean isMandrel(String vendorVersion) {
+            if (vendorVersion == null) {
+                return false;
+            }
+            return !vendorVersion.isBlank() && vendorVersion.startsWith(MANDREL_VERS_PREFIX);
+        }
+
+        private static String mandrelVersion(String vendorVersion) {
+            if (vendorVersion == null) {
+                return null;
+            }
+            int idx = vendorVersion.indexOf(MANDREL_VERS_PREFIX);
+            if (idx < 0) {
+                return null;
+            }
+            String version = vendorVersion.substring(idx + MANDREL_VERS_PREFIX.length());
+            return matchVersion(version);
+        }
+
+        private static String matchVersion(String version) {
+            Matcher versMatcher = VERSION_PATTERN.matcher(version);
+            if (versMatcher.find()) {
+                return versMatcher.group(VERSION_GROUP);
+            }
+            return null;
+        }
+
+        private static String graalVersion(String buildInfo) {
+            if (buildInfo == null) {
+                return null;
+            }
+            int idx = buildInfo.indexOf(JVMCI_BUILD_PREFIX);
+            if (idx < 0) {
+                return null;
+            }
+            String version = buildInfo.substring(idx + JVMCI_BUILD_PREFIX.length());
```

Comment:
This seems fragile as we do not know for how long we will be able to rely on the JVMCI version for figuring out the GraalVM version.

Reply:
Yes, agreed. We'll have to wait and see what the final version will look like. Once we know, I can update this bit.

Reply:
It appears the `jvmci-` bits will no longer be present in more recent GraalVM CE dev builds. As there aren't any available yet, I'll wait for them to show up before I adjust it. It seems the vendor version will give us something going forward.

Reply:
FTR the upstream PR with the latest changes (and an ongoing discussion) is https://github.com/oracle/graal/pull/6356

Reply:
Here would be an alternate approach of using the `release` file (for GraalVM - if we must), fwiw: https://github.com/quarkusio/quarkus/commit/3a9502d2c3d5d20b0053b0bddb24e80034a61f1b

Reply:
We agreed to go with the `jvmci-` bit for now. Resolving conversation.